flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentController.java

```diff
@@ -88,17 +92,23 @@ public DeleteControl cleanup(FlinkDeployment flinkApp, Context context) {
                 operatorNamespace,
                 kubernetesClient,
                 true);
+        jobManagerDeployments.remove(flinkApp);
```

Comment:
Should we also remove from this map when we cancel the flink job in the Flink service?

Reply:
Otherwise after an upgrade the observer might hit the same issue

Reply:
Good catch, will look into that tomorrow.